### PR TITLE
Convert da locale to v2 (closes #1061)

### LINKED ISSUE #1061: Convert da locale to v2
I'll be happy start working on this first thing tomorrow. The [i18n contribution](https://github.com/date-fns/date-fns/blob/master/docs/i18nContributionGuide.md) guide is quite clear so I'll just begin unless there's any objections.

How couldn't I :sweat_smile:  date-fns is super great!
Thank you creators & maintainers for making this awesome project.

### PR DESCRIPTION
**Pull request by MathiasKandelborg**

### CODE REVIEW

**Comment by MathiasKandelborg** [comment]
@kossnocorp This should be mergable now :slightly_smiling_face:

File: src/locale/da/index.js [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
-// }
+var locale = {
+  formatDistance: formatDistance,
+  formatLong: formatLong,
+  formatRelative: formatRelative,
+  localize: localize,
+  match: match,
+  options: {
+    weekStartsOn: 1 /* Monday */,
+    firstWeekContainsDate: 1
```
Comment (mareksuscak) [inline src/locale/da/index.js]: I'm not Dane but I don't think this is correct. In most European countries the first week of the year must contain January 4th so the previous configuration seems to have been correct. Furthermore, according to [this list](http://www.pjh2.de/datetime/weeknumber/wnd.php?l=en) mentioned in [the contribution guide](https://github.com/date-fns/date-fns/blob/master/docs/i18nContributionGuide.md#indexjs), 4 is also the correct number. What do you think?

Comment (MathiasKandelborg) [reply src/locale/da/index.js]: You're right - I don't know what I've been thinking and must've missed it on my latest look through.

File: src/locale/da/_lib/localize/index.js [hunk 2]
```diff
@@ -1,38 +1,162 @@
...
-import buildLocalizeArrayFn from '../../../_lib/buildLocalizeArrayFn/index.js'
 
-var weekdayValues = {
-  narrow: ['sø', 'ma', 'ti', 'on', 'to', 'fr', 'lø'],
-  short: ['søn', 'man', 'tir', 'ons', 'tor', 'fre', 'lør'],
-  long: ['søndag', 'mandag', 'tirsdag', 'onsdag', 'torsdag', 'fredag', 'lørdag']
+var eraValues = {
+  narrow: ['fvt', 'vt'],
+  abbreviated: ['f.v.t.', 'v.t.'],
+  wide: ['før vesterlandsk tidsregning', 'vesterlandsk tidsregning']
```
Comment (Thanerik) [inline src/locale/da/_lib/localize/index.js]: Det er nok nærmere "før vor tidsregning" iht. [ordbogen](https://ordnet.dk/ddo/ordbog?query=f.v.t.) ?

File: src/locale/da/_lib/match/index.js [hunk 3]
```diff
@@ -1,47 +1,136 @@
...
+var parseOrdinalNumberPattern = /\d+/i
 
-var matchWeekdaysPatterns = {
-  narrow: /^(sø|ma|ti|on|to|fr|lø)/i,
-  short: /^(søn|man|tir|ons|tor|fre|lør)/i,
-  long: /^(søndag|mandag|tirsdag|onsdag|torsdag|fredag|lørdag)/i
+var matchEraPatterns = {
+  narrow: /^(fKr|fvt|eKr|vt)/i,
+  abbreviated: /^(f\.Kr\.?|f\.v\.t\.?|e\.Kr\.?|v\.t\.)/i,
+  wide: /^(f.Kr.|før vesterlandsk tidsregning|e.Kr.|vesterlandsk tidsregning)/i
```
Comment (Thanerik) [inline src/locale/da/_lib/match/index.js]: Det er den samme her med "vor tidsregning"

**Comment by MathiasKandelborg** [comment]
@thanerik I do agree, however, it is specified in the unicode Cldr charts which are the universally agreed upon values. We have to take it with Unicode if we want the more actual abbreviation.

Example: https://www.unicode.org/cldr/charts/32/summary/da.html#1639

**Comment by Thanerik** [comment]
I see. I can accept to leave it be "Vestlandersk". As I wrote, I would love to approve your pull request, however I haven't the power to complete it :-) It's a wonderful work you've done!

**Comment by MathiasKandelborg** [comment]
Thanks a lot! I appreciate it.

For now, I've released this PR as [a separate npm package](https://www.npmjs.com/package/Kandelborg-date-fns),as it seems the authors has gone on an OSS vacation :sweat_smile:

**Comment by Thanerik** [comment]
Thank you a lot! It's a great start, until the authors can approve this and merge it into the next beta-release 😄

**Comment by kossnocorp** [comment]
I've released the change as `v2.0.0-alpha.29`: https://gist.github.com/kossnocorp/a307a464760b405bb78ef5020a4ab136#v200-alpha29

**Pull request merged**